src/ucp/core/ucp_request.c

```diff
@@ -194,8 +194,12 @@ ucs_status_t ucp_request_send_buffer_reg(ucp_request_t *req, ucp_lane_index_t la
     status = UCS_OK;
     switch (req->send.datatype & UCP_DATATYPE_CLASS_MASK) {
     case UCP_DATATYPE_CONTIG:
-        status = uct_md_mem_reg(uct_md, (void *)req->send.buffer, req->send.length,
-                                0, &state->dt.contig.memh);
+        if (req->send.length) {
```

Comment:
@yosefe Do we need this check here for `contig`? If yes, I need to add check for `dereg` to avoid memory deregistration in case of `state->dt.contig.memh == UCT_INVALID_MEM_HANDLE`